ntoskrnl/io/pnpmgr/devnode.c

```diff
@@ -0,0 +1,396 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     PnP manager device tree functions
+ * COPYRIGHT:   Casper S. Hornstrup (chorns@users.sourceforge.net)
+ *              2007 Hervé Poussineau (hpoussin@reactos.org)
+ *              2010-2012 Cameron Gutman (cameron.gutman@reactos.org)
+ */
+
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+PDEVICE_NODE IopRootDeviceNode;
```

Comment:
Wow, kernel source code looks weird without the `/* FUNCTIONS ***********/` thing :D

Reply:
If we want this back, good time to do it :D
But honestly I haven't made my mind about these headers

Reply:
Yeah I guess I'd prefer adding these comments here for consistency with the rest of the kernel. Less surprise when reading the code = better ;)

Reply:
What is the difference between `GLOBALS` and `DATA`?

Reply:
I don't think there is a difference. GLOBALS seems more common in ntoskrnl *shrug*